pkg/util/client/client.go

```diff
@@ -25,7 +25,7 @@ import (
 // UpdateFunc is a function that attempts to update a resource.
 // It returns the resulting object, a bool indicating whether an update was made,
 // and an error if any occurred.
```

Comment:
```suggestion
// UpdateFunc is a function that modifies an object in-place.
// It returns a bool indicating whether an update was made, and an error if any occurred.
```

Maybe something like this?